src/ol_concourse/pipelines/infrastructure/k8s_apps/docker_pulumi.py

```diff
@@ -0,0 +1,322 @@
+            GetStep(
+                get=app_image.name,
+                trigger=True,
+                passed=[main_image_build_job.name],
+                params={"skip_download": True},
+            ),
+            LoadVarStep(
+                load_var="image_tag", file=f"{app_image.name}/tag", reveal=True
+            ),
+        ],
+        additional_env_vars={
+            f"{app_name.replace('-', '_').upper()}_DOCKER_TAG": "((.:image_tag))",
+        },
+    )
+
+    # QA and Production Deployments
+    qa_and_production_fragment = pulumi_jobs_chain(
+        pulumi_code=ol_infra_repo,
+        stack_names=[
+            f"applications.{app_name.replace('-', '_')}.QA",
+            f"applications.{app_name.replace('-', '_')}.Production",
+        ],
+        project_name=f"ol-infrastructure-{app_name}-application",
+        project_source_path=(
+            f"src/ol_infrastructure/applications/{app_name.replace('-', '_')}"
+        ),
+        dependencies=[
+            GetStep(
+                get=app_image.name,
+                trigger=True,
+                passed=[rc_image_build_job.name],
+                params={"skip_download": True},
+            ),
+            LoadVarStep(
+                load_var="image_tag", file=f"{app_image.name}/tag", reveal=True
+            ),
+        ],
+        additional_env_vars={
+            f"{app_name.replace('-', '_').upper()}_DOCKER_TAG": "((.:image_tag))",
+        },
+    )
+
+    # Group into Fragments
+
+    main_branch_container_fragement = PipelineFragment(
+        resources=[main_repo, app_image], jobs=[main_image_build_job]
+    )
+
+    release_candidate_container_fragment = PipelineFragment(
+        resources=[
+            release_candidate_repo,
+            # app_image is already included in main_branch_container_fragement
+        ],
+        jobs=[rc_image_build_job],
+    )
+
+    # Consolidate resources for deployment fragments
+    deployment_resources = [
+        ol_infra_repo,
+        pulumi_resource,
+        app_image,
+    ]
+
+    ci_deployment_fragment = PipelineFragment(
+        resource_types=[pulumi_resource_type, *ci_fragment.resource_types],
+        resources=[*deployment_resources, *ci_fragment.resources],
+        jobs=ci_fragment.jobs,
+    )
+
+    # Update qa_and_production_fragment resources similarly if needed,
+    # though pulumi_jobs_chain handles its own resource management internally.
+    # Ensure app_image is available to qa_and_production_fragment jobs.
+    # The dependency GetStep already references app_image.name.
+
+    # Combine all fragments
+
+    combined_fragment = PipelineFragment.combine_fragments(
+        ci_deployment_fragment,
+        qa_and_production_fragment,
+        main_branch_container_fragement,
+        release_candidate_container_fragment,
+    )
+    return combined_fragment.to_pipeline()
+
+
+if __name__ == "__main__":
+    import sys
+
+    app_name = sys.argv[1] if len(sys.argv) > 1 else "learn-ai"
```

Comment:
Is learn ai always what we want as a default unless arguments are specified? I was of the impression this was a k8s wide meta-pipeline?